static/app/components/events/eventReplay/index.tsx

```diff
@@ -1,26 +1,57 @@
+import {useState} from 'react';
+
 import Button from 'sentry/components/button';
 import ErrorBoundary from 'sentry/components/errorBoundary';
 import EventDataSection from 'sentry/components/events/eventDataSection';
 import LazyLoad from 'sentry/components/lazyLoad';
+import {relativeTimeInMs} from 'sentry/components/replays/utils';
 import {t} from 'sentry/locale';
 
 type Props = {
   orgSlug: string;
   projectSlug: string;
   replayId: string;
+  eventTimestamp?: string;
 };
 
-export default function EventReplay({replayId, orgSlug, projectSlug}: Props) {
+export default function EventReplay({
+  replayId,
+  orgSlug,
+  projectSlug,
+  eventTimestamp,
+}: Props) {
+  const [initialTimeOffset, setInitialTimeOffset] = useState<number | undefined>(
+    undefined
+  );
+
+  const getInitialTimeOffset = startTimestampMs => {
```

Comment:
could this be `const initialTimeOffset = relativeTimeInMs(...)`?

Reply:
So the thing here is that I don't have the replay start timestamp to calculate the initial time offset here, that's why I'm passing a callback function - I could move the `EventDataSection` into `ReplayContent` though

Reply:
that might be better. This function is called `get...` but it takes an argument and sets some state.. it's not an intuitive function to me and will be hard to test.